pandas/tests/groupby/test_function.py

```diff
@@ -641,10 +641,10 @@ def test_groupby_cumprod():
     tm.assert_series_equal(actual, expected)
 
     df = DataFrame({"key": ["b"] * 100, "value": 2})
+    df["value"] = df["value"].astype(float)
```

Comment:
We can maybe keep this with as int (or test both in addition), so we have a test for the silent overflow behaviour?

Reply:
Added a new test explicitly testing that overflow is consistent with numpy